feature/account/setup/src/main/kotlin/app/k9mail/feature/account/setup/ui/validation/AccountValidationViewModel.kt

```diff
@@ -0,0 +1,143 @@
+import app.k9mail.feature.account.setup.ui.validation.AccountValidationContract.State
+import app.k9mail.feature.account.setup.ui.validation.AccountValidationContract.ViewModel
+import com.fsck.k9.mail.server.ServerSettingsValidationResult
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+private const val CONTINUE_NEXT_DELAY = 1000L
+
+internal class AccountValidationViewModel(
+    initialState: State = State(),
+    private val validateServerSettings: DomainContract.UseCase.ValidateServerSettings,
+) : BaseViewModel<State, Event, Effect>(initialState), ViewModel {
+
+    override fun initState(state: State) {
+        updateState {
+            state.copy(
+                isLoading = false,
+                error = null,
+                isSuccess = false,
+            )
+        }
+    }
+
+    override fun event(event: Event) {
+        when (event) {
+            Event.ValidateServerSettings -> onValidateConfig()
+            Event.OnNextClicked -> TODO()
+            Event.OnBackClicked -> onBack()
+            Event.OnRetryClicked -> onRetry()
+        }
+    }
+
+    private fun onValidateConfig() {
+        if (state.value.isSuccess) {
+            navigateNext()
+        } else {
+            validateServerSettings()
+        }
+    }
+
+    private fun validateServerSettings() {
+        viewModelScope.launch {
+            val serverSettings = state.value.serverSettings
+            if (serverSettings == null) {
+                updateError(Error.UnknownError("Server settings not set"))
+                return@launch
+            }
+
+            updateState {
+                it.copy(isLoading = true)
+            }
+
+            when (val result = validateServerSettings.execute(serverSettings)) {
+                ServerSettingsValidationResult.Success -> updateSuccess()
+
+                is ServerSettingsValidationResult.AuthenticationError -> updateError(
+                    Error.AuthenticationError(result.serverMessage),
+                )
+
+                is ServerSettingsValidationResult.CertificateError -> updateError(
+                    Error.CertificateError(result.certificateChain),
+                )
+
+                is ServerSettingsValidationResult.NetworkError -> updateError(
+                    Error.NetworkError(result.exception),
+                )
+
+                is ServerSettingsValidationResult.ServerError -> updateError(
+                    Error.ServerError(result.serverMessage),
+                )
+
+                is ServerSettingsValidationResult.UnknownError -> updateError(
+                    Error.UnknownError(result.exception.message ?: "Unknown error"),
+                )
+            }
+        }
+    }
+
+    private fun updateSuccess() {
+        updateState {
+            it.copy(
+                isLoading = false,
+                isSuccess = true,
+            )
+        }
+
+        viewModelScope.launch {
```

Comment:
This coroutine needs to be canceled when the user navigates back.

Reply:
Fixed in a follow up pull-request